api/src/main/java/org/apache/cloudstack/api/command/admin/network/CreateNetworkCmdByAdmin.java

```diff
@@ -42,6 +42,12 @@
     @Parameter(name=ApiConstants.HIDE_IP_ADDRESS_USAGE, type=CommandType.BOOLEAN, description="when true ip address usage for the network will not be exported by the listUsageRecords API")
     private Boolean hideIpAddressUsage;
 
+    @Parameter(name = ApiConstants.ROUTER_IP, type = CommandType.STRING, description = "IPV4 address to be assigned to a router in a shared network", since = "4.16")
+    private String routerIp;
+
+    @Parameter(name = ApiConstants.ROUTER_IPV6, type = CommandType.STRING, description = "IPV6 address to be assigned to a router in a shared network", since = "4.16")
+    private String routerIpv6;
```

Comment:
can add validator "_ApiArgValidator.NotNullOrEmpty_" for both args ?

Reply:
Wouldn't  adding such a validator make the field/ parameter mandatory?

Reply:
> Wouldn't adding such a validator make the field/ parameter mandatory?

I think it is for validating the parameter value, nothing to do with required - true/false.